packages/replayio/src/utils/protocol/api/ensureProcessed.ts

```diff
@@ -0,0 +1,39 @@
+import { ensureProcessedParameters, ensureProcessedResult } from "@replayio/protocol";
+import ProtocolClient from "../ProtocolClient";
+import { debug } from "../debug";
+
+export async function ensureProcessed(client: ProtocolClient, sessionId: string) {
+  await client.waitUntilAuthenticated();
+
+  let removeListeners: Function[] = [];
+
+  return new Promise<void>(async (resolve, reject) => {
+    try {
+      removeListeners.push(
+        client.listenForMessage("Recording.sessionError", (error: any) => {
+          reject(error);
```

Comment:
nit: A less indented and more linear~ flow could be easier to follow. A deferred resolved here racing with that `Session.ensureProcessed` would spare you the need to construct `new Promise` manually in this function